src/test/ui/unboxed-closures/unboxed-closures-failed-recursive-fn-2.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0282]: type annotations needed
+error[E0282]: type annotations needed in `std::option::Option<T>`
   --> $DIR/unboxed-closures-failed-recursive-fn-2.rs:16:32
    |
 LL |     let mut closure0 = None;
-   |         ------------ consider giving `closure0` a type
+   |         ------------ consider giving `closure0` the type `std::option::Option<T>` with the type parameter `_` specified
```

Comment:
I think it's a problem when the type parameters don't match up. (It'd be better for both to be `_`, as long as it's unambiguous.)

Reply:
I believe what is happening here is that `_` and `T` are actually not unified, they are _separate_ types as far as `typeck` is concerned. I'm not sure what the best way to go about it here would be :-/

Reply:
Cheated a bit and changed it to be:

```
error[E0282]: type annotations needed for `std::option::Option<T>`
  --> $DIR/unboxed-closures-failed-recursive-fn-2.rs:16:32
   |
LL |     let mut closure0 = None;
   |         ------------ consider giving `closure0` the explicit type `std::option::Option<T>`, with the type parameters specified
...
LL |                         return c();
   |                                ^^^ cannot infer type
   |
   = note: type must be known at this point
```

Reply:
Yeah, that's probably better.